recipes/llvm-core/all/conanfile.py

```diff
@@ -0,0 +1,283 @@
+        cmake.definitions['LLVM_ENABLE_WERROR'] = False
+
+        cmake.definitions['LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN'] = True
+        cmake.definitions['LLVM_USE_RELATIVE_PATHS_IN_DEBUG_INFO'] = False
+        cmake.definitions['LLVM_BUILD_INSTRUMENTED_COVERAGE'] = False
+        cmake.definitions['LLVM_OPTIMIZED_TABLEGEN'] = True
+        cmake.definitions['LLVM_REVERSE_ITERATION'] = False
+        cmake.definitions['LLVM_ENABLE_BINDINGS'] = False
+        cmake.definitions['LLVM_CCACHE_BUILD'] = False
+
+        cmake.definitions['LLVM_INCLUDE_TOOLS'] = self.options.shared
+        cmake.definitions['LLVM_INCLUDE_EXAMPLES'] = False
+        cmake.definitions['LLVM_INCLUDE_TESTS'] = False
+        cmake.definitions['LLVM_INCLUDE_BENCHMARKS'] = False
+        cmake.definitions['LLVM_APPEND_VC_REV'] = False
+        cmake.definitions['LLVM_BUILD_DOCS'] = False
+        cmake.definitions['LLVM_ENABLE_IDE'] = False
+
+        cmake.definitions['LLVM_ENABLE_EH'] = self.options.exceptions
+        cmake.definitions['LLVM_ENABLE_RTTI'] = self.options.rtti
+        cmake.definitions['LLVM_ENABLE_THREADS'] = self.options.threads
+        cmake.definitions['LLVM_ENABLE_LTO'] = self.options.lto
+        cmake.definitions['LLVM_STATIC_LINK_CXX_STDLIB'] = \
+            self.options.static_stdlib
+        cmake.definitions['LLVM_ENABLE_UNWIND_TABLES'] = \
+            self.options.unwind_tables
+        cmake.definitions['LLVM_ENABLE_EXPENSIVE_CHECKS'] = \
+            self.options.expensive_checks
+        cmake.definitions['LLVM_ENABLE_ASSERTIONS'] = \
+            self.settings.build_type == 'Debug'
+
+        cmake.definitions['LLVM_USE_NEWPM'] = False
+        cmake.definitions['LLVM_USE_OPROFILE'] = False
+        cmake.definitions['LLVM_USE_PERF'] = self.options.use_perf
+
+        cmake.definitions['LLVM_ENABLE_Z3_SOLVER'] = False
+        cmake.definitions['LLVM_ENABLE_LIBPFM'] = False
+        cmake.definitions['LLVM_ENABLE_LIBEDIT'] = False
+        cmake.definitions['LLVM_ENABLE_FFI'] = self.options.with_ffi
+        cmake.definitions['LLVM_ENABLE_ZLIB'] = \
+            self.options.get_safe('with_zlib', False)
+        cmake.definitions['LLVM_ENABLE_LIBXML2'] = \
+            self.options.get_safe('with_xml2', False)
+        return cmake
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+            del self.options.with_zlib
+            del self.options.with_xml2
+
+    def requirements(self):
+        if self.options.with_ffi:
+            self.requires('libffi/3.3')
+        if self.options.get_safe('with_zlib', False):
+            self.requires('zlib/1.2.11')
+        if self.options.get_safe('with_xml2', False):
+            self.requires('libxml2/2.9.10')
+
+    def configure(self):
+        if self.settings.os == 'Windows' and self.options.shared:
+            raise ConanInvalidConfiguration('shared lib not supported')
```

Comment:
```suggestion
            raise ConanInvalidConfiguration('Shared build is not supported on Windows')
```

Reply:
Already done in another way. I like to stick to **PEP8** and the line was too long.